MalApi/MyAnimeListApi.cs

```diff
@@ -227,6 +305,147 @@ public Task<MalUserLookupResults> GetAnimeListForUserAsync(string user)
         }
 
         /// <summary>
+        /// Updates a user's anime list entry.
+        /// </summary>
+        /// <param name="animeId">ID of the anime</param>
+        /// <param name="updateInfo">The updated information</param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="user">MAL user</param>
+        /// <param name="password">MAL password</param>
+        /// <returns></returns>
+        public async Task<string> UpdateAnimeForUserAsync(int animeId, UpdateAnimeObject updateInfo, string user, string password, CancellationToken cancellationToken)
+        {
+            string userInfoUri = string.Format(MalAppAnimeUpdateUri, Uri.EscapeDataString(animeId.ToString()));
+
+            Logging.Log.InfoFormat("Updating anime entry for MAL anime ID {0}, user {1} using URI {2}", animeId, user, userInfoUri);
+
+            Func<string, string> responseProcessingFunc = (response) =>
+            {
+                using (TextReader xmlTextReader = new StringReader(response))
+                {
+                    try
+                    {
+                        return xmlTextReader.ToString();
+                    }
+                    catch (Exception ex)
+                    {
+                        throw new Exception(string.Format("Exception for user {0} and anime ID {1}.", user, animeId), ex);
+                    }
```

Comment:
responseProcessingFunc should not be catching exceptions and rethrowing them as a general Exception. The baseErrorMessage parameter to ProcessRequestAsync is used as part of the error handling in ProcessRequestAsync.

Reply:
https://myanimelist.net/modules.php?go=api says the response is 'Updated' or detailed error message. Have you tested this? If there is an error does it return an HTTP error status code or does it return an OK HTTP status code? If there is an error message, is it a plain string or is it in something like xml or json? It looks like the code here does nothing to surface the MAL error message to the caller.